realm/realm-library/src/androidTest/java/io/realm/RealmObjectSchemaTests.java

```diff
@@ -194,6 +224,24 @@ public boolean isNullable() {
 
     @Test
     public void addRemoveField() {
+        if (type == ObjectSchemaType.IMMUTABLE) {
+            try {
+                schema.addField("test", int.class);
```

Comment:
the test is called `addRemove()` but you only test `add*` for the immutable version. Also add a test for `removeField`?